types/@ember/object/core.d.ts

```diff
@@ -0,0 +1,7 @@
+import _CoreObject from 'ember__object/core';
+export * from 'ember__object/core';
+
+export default class CoreObject extends _CoreObject {
+  // https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31481
+  init(...args: any): void;
```

Comment:
@josemarluedke Since you made the original comment about `super.init(...arguments)` in a prior commit, you might be interested in the discussion in the https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31481 and on [Discord](https://discordapp.com/channels/480462759797063690/484421406659182603/524495974584811532).

Reply:
This is interesting. What I got form that conversation was that this isn't really necessary today as we don't have any args in Ember. However, it does keep it compatible if Ember would add a new argument. So for apps, this might not be necessary, but to keep the addon future proof, this is okay.

Is the above summary correct?

Reply:
Adding `...arguments` would fails the type check for addons. So without adding manual type overrides like above, this wouldn't fly. It is also not very likely, that existing framework hooks would receive new arguments in the future.

What we should do here is remove the above type definition and the `...arguments`.